File: fix_turner.py
import os
import re
from pathlib import Path

base_dir = Path("reference/turner/turnerasphalt.com")

# Tracking patterns to remove
tracking_patterns = [
    r'<script[^>]*google-analytics[^>]*>.*?</script>',
    r'<script[^>]*googletagmanager[^>]*>.*?</script>',
    r'<script[^>]*gtag[^>]*>.*?</script>',
    r'<script[^>]*fbevents[^>]*>.*?</script>',
    r'<script[^>]*facebook[^>]*>.*?</script>',
    r"<script[^>]*>.*?gtag\(.*?</script>",
    r"<script[^>]*>.*?dataLayer.*?</script>",
    r'<noscript>.*?googletagmanager.*?</noscript>',
    r'<noscript>.*?facebook.*?</noscript>',
]

# First, rename files with @ver= to remove the query string part
print("Renaming files with query strings...")
renamed_files = {}
for root, dirs, files in os.walk(base_dir):
    for file in files:
        if '@' in file:
            old_path = Path(root) / file
            # Extract base name without query string
            new_name = file.split('@')[0]
            # Handle .css and .js extensions
            if not new_name.endswith(('.css', '.js', '.woff2', '.woff', '.ttf', '.png', '.jpg', '.webp', '.svg')):
                ext = file.split('.')[-1] if '.' in file.split('@')[-1] else ''
                if ext:
                    new_name = new_name + '.' + ext
            new_path = Path(root) / new_name

            # Track the rename for URL fixing
            old_relative = str(old_path.relative_to(base_dir)).replace('\\', '/')
            new_relative = str(new_path.relative_to(base_dir)).replace('\\', '/')
            renamed_files[old_relative] = new_relative

            if not new_path.exists():
                print(f"  {file} -> {new_name}")
                old_path.rename(new_path)
            else:
                # If target exists, just delete the @ver version
                old_path.unlink()

print(f"\nRenamed {len(renamed_files)} files")

# Now fix HTML files
print("\nFixing HTML files...")
html_files = list(base_dir.glob("**/*.html"))

for html_file in html_files:
    print(f"Processing: {html_file.name}")
    content = html_file.read_text(encoding='utf-8', errors='ignore')
    original = content

    # Remove tracking scripts
    for pattern in tracking_patterns:
        content = re.sub(pattern, '', content, flags=re.DOTALL | re.IGNORECASE)

    # Fix URLs with query strings - convert ?ver=xxx to nothing
    content = re.sub(r'\?ver=[^"\'&\s]+', '', content)
    content = re.sub(r'&ver=[^"\'&\s]+', '', content)

    # Fix references to @ver files
    content = re.sub(r'@ver=[^"\'&\s]+\.css', '.css', content)
    content = re.sub(r'@ver=[^"\'&\s]+\.js', '.js', content)

    # Remove external links to turnerasphalt.com and make them relative
    content = re.sub(r'https?://turnerasphalt\.com/?', '/', content)
    content = re.sub(r'https?://www\.turnerasphalt\.com/?', '/', content)

    # Remove Google fonts external links (keep local copies if they exist)
    # content = re.sub(r'<link[^>]*fonts\.googleapis\.com[^>]*>', '', content)

    if content != original:
        html_file.write_text(content, encoding='utf-8')
        print(f"  Updated: {html_file.name}")

# Also fix CSS files that might have url() references
print("\nFixing CSS files...")
css_files = list(base_dir.glob("**/*.css"))
for css_file in css_files:
    try:
        content = css_file.read_text(encoding='utf-8', errors='ignore')
        original = content

        # Fix URLs with query strings
        content = re.sub(r'\?ver=[^"\')\s]+', '', content)

        if content != original:
            css_file.write_text(content, encoding='utf-8')
            print(f"  Updated: {css_file.name}")
    except:
        pass

print("\nDone! Turner site should now work locally.")

File: fix_turner.js
const fs = require('fs');
const path = require('path');

const baseDir = path.join(__dirname, 'reference/turner/turnerasphalt.com');

// Tracking patterns to remove
const trackingPatterns = [
    /<script[^>]*google-analytics[^>]*>[\s\S]*?<\/script>/gi,
    /<script[^>]*googletagmanager[^>]*>[\s\S]*?<\/script>/gi,
    /<script[^>]*gtag[^>]*>[\s\S]*?<\/script>/gi,
    /<script[^>]*fbevents[^>]*>[\s\S]*?<\/script>/gi,
    /<script[^>]*>[\s\S]*?gtag\([\s\S]*?<\/script>/gi,
    /<script[^>]*>[\s\S]*?dataLayer[\s\S]*?<\/script>/gi,
    /<noscript>[\s\S]*?googletagmanager[\s\S]*?<\/noscript>/gi,
    /<noscript>[\s\S]*?facebook[\s\S]*?<\/noscript>/gi,
];

function walkDir(dir, callback) {
    const files = fs.readdirSync(dir);
    files.forEach(file => {
        const filepath = path.join(dir, file);
        const stat = fs.statSync(filepath);
        if (stat.isDirectory()) {
            walkDir(filepath, callback);
        } else {
            callback(filepath);
        }
    });
}

// Step 0: Fix double extensions (.css.css -> .css, .js.js -> .js)
console.log("Fixing double extensions...");
let doubleExtFixed = 0;

walkDir(baseDir, (filepath) => {
    const filename = path.basename(filepath);
    const dir = path.dirname(filepath);

    if (filename.endsWith('.css.css')) {
        const newName = filename.slice(0, -4); // Remove the extra .css
        const newPath = path.join(dir, newName);

        if (!fs.existsSync(newPath)) {
            console.log(`  ${filename} -> ${newName}`);
            fs.renameSync(filepath, newPath);
            doubleExtFixed++;
        } else {
            // Delete duplicate
            fs.unlinkSync(filepath);
            doubleExtFixed++;
        }
    } else if (filename.endsWith('.js.js')) {
        const newName = filename.slice(0, -3); // Remove the extra .js
        const newPath = path.join(dir, newName);

        if (!fs.existsSync(newPath)) {
            console.log(`  ${filename} -> ${newName}`);
            fs.renameSync(filepath, newPath);
            doubleExtFixed++;
        } else {
            // Delete duplicate
            fs.unlinkSync(filepath);
            doubleExtFixed++;
        }
    }
});

console.log(`\nFixed ${doubleExtFixed} double extensions`);

// Step 1: Rename files with @ in the name
console.log("\nRenaming files with query strings...");
let renamedCount = 0;

walkDir(baseDir, (filepath) => {
    const filename = path.basename(filepath);
    if (filename.includes('@')) {
        const dir = path.dirname(filepath);
        const baseName = filename.split('@')[0];

        // Get the extension from the base name (before @)
        // e.g., "style.css@ver=1.8.0.css" -> baseName is "style.css"
        // If baseName already has extension, use it; otherwise check after @
        let newName = baseName;

        // Only add extension if baseName doesn't already have one
        if (!baseName.match(/\.(css|js|woff2?|ttf|eot|svg|png|jpg|gif)$/i)) {
            const afterAt = filename.split('@')[1] || '';
            if (afterAt.includes('.css')) newName += '.css';
            else if (afterAt.includes('.js')) newName += '.js';
            else if (afterAt.includes('.woff2')) newName += '.woff2';
            else if (afterAt.includes('.woff')) newName += '.woff';
        }

        const newPath = path.join(dir, newName);

        if (!fs.existsSync(newPath)) {
            console.log(`  ${filename} -> ${newName}`);
            fs.renameSync(filepath, newPath);
            renamedCount++;
        } else {
            // Delete duplicate
            fs.unlinkSync(filepath);
        }
    }
});

console.log(`\nRenamed ${renamedCount} files`);

// Step 2: Fix HTML files
console.log("\nFixing HTML files...");

walkDir(baseDir, (filepath) => {
    if (!filepath.endsWith('.html')) return;

    let content = fs.readFileSync(filepath, 'utf8');
    const original = content;

    // Remove tracking scripts
    trackingPatterns.forEach(pattern => {
        content = content.replace(pattern, '');
    });

    // Fix URLs with query strings
    content = content.replace(/\?ver=[^"'&\s]+/g, '');
    content = content.replace(/&ver=[^"'&\s]+/g, '');

    // Fix references to @ver files
    content = content.replace(/@ver=[^"'&\s]+\.css/g, '.css');
    content = content.replace(/@ver=[^"'&\s]+\.js/g, '.js');
    content = content.replace(/@ver=[^"'&\s]+/g, '');

    // Make turnerasphalt.com links relative
    content = content.replace(/https?:\/\/(www\.)?turnerasphalt\.com\/?/g, '/');

    if (content !== original) {
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`  Updated: ${path.basename(filepath)}`);
    }
});

// Step 3: Fix CSS files
console.log("\nFixing CSS files...");

walkDir(baseDir, (filepath) => {
    if (!filepath.endsWith('.css')) return;

    try {
        let content = fs.readFileSync(filepath, 'utf8');
        const original = content;

        content = content.replace(/\?ver=[^"')\s]+/g, '');

        if (content !== original) {
            fs.writeFileSync(filepath, content, 'utf8');
            console.log(`  Updated: ${path.basename(filepath)}`);
        }
    } catch (e) {}
});

console.log("\nDone! Turner site should now work locally.");
